src/bot/commands/utilities/forecast.ts

```diff
@@ -0,0 +1,296 @@
+import { Command, Input } from '@api';
+import * as request from 'request';
+import { Message } from 'discord.js';
+import { Emoji } from '@bot/libraries/emoji';
+import { RichEmbed } from 'discord.js';
+import { Framework } from '@core/framework';
+import * as moment from 'moment';
+
+export class Forecast extends Command {
+    constructor() {
+        super({
+            name: 'forecast',
+            description: 'Checks the weather forecast at a given location.',
+            arguments: [
+                {
+                    name: 'location',
+                    description: 'The name of a city to search for.',
+                    expand: true,
+                    required: true
+                }
+            ]
+        });
+    }
+
+    async execute(input: Input) {
+        let location = input.getArgument('location') as string;
+
+        // Error if there is no API key
+        if (!Framework.getConfig().authentication.openWeatherMap) {
+            return await input.channel.send(`${Emoji.ERROR}  This bot is not configured for weather.`);
+        }
+
+        // Get the key
+        let key = Framework.getConfig().authentication.openWeatherMap.key;
+
+        // Show a loading message
+        let message = await input.channel.send(`${Emoji.LOADING}  Fetching weather...`) as Message;
+
+        // Get the current weather
+        let today : Day;
+        try {
+            today = await this.getCurrent(location, key);
+        }
+        catch (err) {
+            await message.edit(`${Emoji.ERROR}  ${err.message}`);
```

Comment:
We need to stop assuming that the messages exists, because people can delete them if we're not careful.